### DISCUSSION TITLE
UX/UI: Adding Application > Config > Name to titlebar

### DISCUSSION THREAD

**Discussion by Gatada** [godotengine/godot-proposals#5442] [Editor]
I would like to see the name of the project in the titlebar of the Godot Main Window (ref. mockup).

"Name" refers to the string set in Project Settings > Application > Config > Name.

<img width="583" alt="Screenshot 2022-09-17 at 12 56 19" src="https://user-images.githubusercontent.com/326334/190855554-b2cdbde4-5831-4cc1-9cc6-27ca370c40ff.png">

When no name is set, it would say Untitled or Unnamed.

I suggest to the far left, as it is currently unoccupied.

**Comment by Calinou** [comment]
This was already changed for better usability last year: https://github.com/godotengine/godot/pull/47673

The scene name is shown first, as this is the most "local" information.

**Reply by Gatada** [reply]
Thanks. I just don't see that in 4.0b1. I guess it's a bug then? Or am I doing something wrong?

**Reply by Calinou** [reply]
Which operating system are you using? Can you take a screenshot of what the window title looks like on your system?

**Reply by Gatada** [reply]
I'm on macOS. Earlier I used 12.5 and now I am on 12.6 - the Godot window and menubar looks the same:

<img width="2560" alt="Screenshot 2022-09-19 at 11 01 41" src="https://user-images.githubusercontent.com/326334/191002513-b0ca5546-f286-49b5-b639-6d46c12b8c9b.png">

**Reply by Calinou** [reply]
I didn't think about the recent macOS global menu integration, so that's why I couldn't reproduce this on my end.

@bruvzg Do you know if the project name could be written in the empty space between the top-left close buttons and the 2D/3D/Script/AssetLib buttons? It may have to use ellipsis if the window is small though.

**Reply by bruvzg** [reply]
The default title bar text would interfere with the `2D/3D/...` tabs, but I guess we can add a custom one to the part of the title in between.

**Comment by Gatada** [comment]
So, it is a genuine issue. My first non-financial contribution to Godot!

What do I do now? Do I create an issue or something - or do I leave it with this discussion?

**Reply by Calinou** [reply]
You can leave this discussion as-is for now. Someone may eventually get around to implementing this, but it's not high-priority as the goal is currently to stabilize 4.0 for release.

**Comment by akien-mga** [comment]
Implemented by https://github.com/godotengine/godot/pull/66254.

**Comment by Gatada** [comment]
So now this discussion is closed? .. I just can't do it Captain, I don't have the Power!

**Reply by Calinou** [reply]
> So now this discussion is closed? .. I just can't do it Captain, I don't have the Power!

Yes, but we don't lock proposal discussions unless they're a duplicate of another proposal.
